src/main/java/at/hannibal2/skyhanni/features/inventory/experimentationtable/SuperpairDataDisplay.kt

```diff
@@ -0,0 +1,257 @@
+            add(" $prefix §e${match.key.pair.first.reward}")
+        }
+        if (powerups.isNotEmpty()) add("§bPowerUp")
+        for (powerup in powerups) {
+            val prefix = determinePrefix(powerups.indexOf(powerup), powerups.size - 1)
+            add(" $prefix §b${powerup.key.item.reward}")
+        }
+        val toAdd = mutableListOf<String>()
+        if (possiblePairs >= 1) toAdd.add("§ePairs - $possiblePairs")
+        if (2 - powerups.size >= 1) toAdd.add("§bPowerUps - ${2 - powerups.size}")
+        if (normals.isNotEmpty()) toAdd.add("§7Normals - ${normals.size}")
+
+        if (toAdd.isNotEmpty()) {
+            add("")
+            add("§4Not found")
+        }
+        for (string in toAdd) if (string != toAdd.last()) add(" ├ $string") else add(" └ $string")
+    }
+
+    private fun calculatePossiblePairs(currentExperiment: Experiment) =
+        ((currentExperiment.gridSize - 2) / 2) - found.filter { it.value != FoundType.POWERUP }.size
+
+    private fun convertToReward(item: ItemStack) = if (item.displayName.removeColor() == "Enchanted Book") item.getLore()[2].removeColor()
+    else item.displayName.removeColor()
+
+    private fun determinePrefix(index: Int, lastIndex: Int) = if (index == lastIndex) "└" else "├"
+
+    private fun hasFoundPair(
+        pair: ItemPair,
+    ) = pair.first.index != pair.second.index && pair.first.sameAs(pair.second)
+
+    private fun hasFoundMatch(firstItem: Item) =
+        uncoveredItems.any { it.value.index != firstItem.index && it.value.sameAs(firstItem) } &&
+            found.none {
+                it.value.isOneOf(
+                    FoundType.PAIR,
+                    FoundType.MATCH,
+                ) && anyDuplicates(firstItem.index, it.key.pair.first.index, it.key.pair.second.index)
+            }
+
+    private fun isPowerUp(reward: String) = ExperimentationTableAPI.powerUpPattern.matches(reward)
+
+    private fun isReward(reward: String) =
+        ExperimentationTableAPI.rewardPattern.matches(reward) || ExperimentationTableAPI.powerUpPattern.matches(reward)
+
+    private fun isWaiting(itemName: String) =
+        listOf("Click any button!", "Click a second button!", "Next button is instantly rewarded!").contains(itemName)
+
+    private fun clicksSinceSeparator(list: MutableMap<Int, Item>): Int {
+        val lastIndex = list.entries.indexOfLast { it.value == emptyItem }
+        return if (lastIndex != -1) list.size - 1 - lastIndex else -1
+    }
+
+    private fun isOutOfBounds(slot: Int, experiment: Experiment): Boolean =
+        slot <= experiment.startSlot || slot >= experiment.endSlot ||
+            (if (experiment.sideSpace == 1) slot in sideSpaces1 else slot in sideSpaces2)
+
+    private fun anyDuplicates(search: Any, vararg toSearch: Any) = search in toSearch
+
+    private fun Item.sameAs(other: Item) =
+        this.reward == other.reward && this.damage == other.damage
```

Comment:
is the only use case of `damage` to get compared here? why cant we work with internal name instead of with item name in reward? should make the damage member useless

Reply:
this is so I can compare dyes

Reply:
oh that makes sense. i thought you were comparing the different xp values via item name. (does the xp value stand in the lore?, is there a chance two pairs have the same xp value but different colors?)

Reply:
the xp value is the item name and they can be the same xp value with different colors, yes

Reply:
does the item name color stay the same for different pairs? if so, this is sad and your damage approach is fine (maybe needs rephrase or clarification why this exist)

Reply:
yes its the same color for every xp pair